api/auth/authenticate.js

```diff
@@ -1,10 +1,33 @@
-module.exports = (username, password, done) => {
-  // Check the username and password.  If it's good, callback
-  // with a valid user object.
-  if (username === 'hello' && password === 'world') {
-    done(null, { username, email: 'hello@world.com', id: 'user-id' });
-  } else {
-    // Otherwise, callback with an error.
-    done('Unknown user');
-  }
-};
+const defaultDB = require('../db')();
+const defaultBcrypt = require('bcryptjs');
+
+module.exports = (
+  db = defaultDB,
+  bcrypt = defaultBcrypt
+) => ((username, password, done) => db('users')
+  .where({
+    email: username
+  })
+  .select()
+  .then((users) => {
+    let user = { password: '' };
```

Comment:
thoughts on early exiting if no `users` are found (before checking if passwords match)? i guess it means repeating the "failed login" callback; eh, either way is fine i suppose. also not sure how you feel about ternaries ;)

`const user = users.length ? users[0] : { password: '' }`